### ISSUE TITLE
[bug] The very first output message fails to print a home_dir

### ISSUE THREAD

**Issue by belkka** [GothenburgBitFactory/taskwarrior#2240]
#### To report a bug...

* What command(s) did you run?
`task`

* What did you expect to happen?
`A configuration file could not be found in ~`
This is what is supposed to be print according to https://taskwarrior.org/docs/introduction.html, but I actually accept ANY non-empty string after "found in ". I do not expect EXACTLY `~`, but it must not be empty.

* What actually happened?
`A configuration file could not be found in ` ← no symbols after "in "
Note: everything works if I type "yes" and create a sample config file. But I just installed the program and its the very first output message looks confusing. It says that it failed to found a configuration in some directory, but it fails to output that directory. This first impression makes me feel that program is broken, despite of being rather a cosmetic bug.

* Paste the output of the ```task diag``` command.
```
$ LANG=C task diag
A configuration file could not be found in 

Would you like a sample /home/ulianich/.taskrc created, so Taskwarrior can proceed? (yes/no) no
Cannot proceed without rc file.
```
I hope this helps :)

---
Here is my own investigation on the problem:
1. The search for ["A configuration file could not be found"](https://github.com/GothenburgBitFactory/taskwarrior/search?q="A+configuration+file+could+not+be+found"&unscoped_q="A+configuration+file+could+not+be+found") string shows that the only place with such a message is https://github.com/GothenburgBitFactory/taskwarrior/blob/f6b2a6541c462eab1e89c3bbcb4610eca84ec295/src/Context.cpp#L1112
The global variable `home_dir` is substituted after "in "

2. The search for [home_dir](https://github.com/GothenburgBitFactory/taskwarrior/search?q=home_dir&unscoped_q=home_dir) string shows that:
  - `home_dir` is `std::string` with default value being empty
  - the only line where it can potentially be set is https://github.com/GothenburgBitFactory/taskwarrior/blob/f6b2a6541c462eab1e89c3bbcb4610eca84ec295/src/Context.cpp#L457 which is a call to `CLI2::getOverride()`

3. According to implementation of `getOverride()`
https://github.com/GothenburgBitFactory/taskwarrior/blob/7c7400b3679262c7f5cce20a7a6b6ac7aedd246a/src/CLI2.cpp#L239-L250 the `home` variable could be left untouched (keeping its empty value) if `getValue (argc, argv, "rc")` returns `nullptr`

4. According to implementation of `getValue()` https://github.com/GothenburgBitFactory/taskwarrior/blob/7c7400b3679262c7f5cce20a7a6b6ac7aedd246a/src/CLI2.cpp#L217-L235 this can happen if command line arguments (i. e. `argv` string array) does NOT contain any argument matching the `rc:.*` pattern. Just to check this, I have run the following command:
```
$ task rc:xxx
A configuration file could not be found in /home/ulianich

Would you like a sample /home/ulianich/xxx created, so Taskwarrior can proceed? (yes/no) no
Using alternate .taskrc file /home/ulianich/xxx
Cannot proceed without rc file.
```
which now works as expected! (although I do not understand the meaning of "rc:..." parameter and don't understand why isn't it kind of `--rc ...` instead)

Note: "/home/ulianich" is my `$HOME` environment variable in all commands above.

---
I am not sure what is correct way to fix the bug. Since example at https://taskwarrior.org/docs/introduction.html is correct, I assume that it used to be OK and then something wrong happen, so it become broken. Maybe `home_dir` default value must be `"~"` instead of empty string?

**Comment by tbabej** [comment]
@belkka Thanks for the extensive bug report. I can confirm this happens on 2.5.2.

**Comment by tbabej** [comment]
Fixed on 2.6.0:

```
[root@tw test]# task
A configuration file could not be found in /root

Would you like a sample /root/.taskrc created, so Taskwarrior can proceed? (yes/no)
```
